### ISSUE TITLE
tang: /usr/sbin/tangd fails with "Illegal instruction" in openwrt 23.05

### ISSUE THREAD

**Issue by lespinasse** [openwrt/packages#21579]
Maintainer: @Tiboris @neheb @cotequeiroz

Environment: Openwrt 23.05.0-rc2 running on a raspberry pi zero W

Maintainer: @\<github-user> (find it by checking history of the package Makefile)
Environment: (put here arch, model, OpenWrt version)

Description:

I have been setting up a couple tang instances on raspberry pi zero W hardware. With openwrt 22.03, tang seems to work fine.

I wanted to try with openwrt 23.05 (I was interested in the standalone feature from tang 12), but I have not been able to get tang going there.

After setting `option enabled '1'` in /etc/config/tang, the `/etc/init.d/tang start` script fails silently (the tang port is not opened for listening).

Running `/usr/sbin/tangd -p 9090 -l /usr/share/tang/db` manually, I see:

```
# /usr/sbin/tangd -p 9090 -l /usr/share/tang/db
Listening on 0.0.0.0:9090
Listening on [::]:9090
Illegal instruction
#
```

**Comment by lespinasse** [comment]
Looking into it a little longer, I was able to get tang to work on openwrt 23.05 by running it with xinetd rather than standalone. So, the issue seems to be with the standalone code that was added in tang version 12. Wild guess, maybe it might be the issue that was fixed by https://github.com/latchset/tang/commit/9b9d14973aa1e5b2290595a6b53e67c2f4e0369f in tang version 13.

**Comment by neheb** [comment]
Linked issue sounds like it's for 32-bit.

**Comment by brada4** [comment]
ccan you extract relevant lines from logread/dmesg? It should jave backtrace and pinpoint to instruction in a source file.

**Comment by lespinasse** [comment]
@brada4 logread / dmesg do not show anything about the crash. Also I don't think java should be involved as tang is written in C ?

@neheb That makes sense, pi zero W is a 32 bit platform.

**Comment by lespinasse** [comment]
@neheb I can confirm tangd works fine in a standalone setup on pi 4B, which is a 64 bit platform. So I guess that confirms the theory about the issue being exposed on 32 bit systems only.

**Comment by samcday** [comment]
I just ran into this issue with my AVM FRITZ!Box 4040 (ipq4xxx / armv7l) when upgrading from 22.03.3 to 22.03.5, so this doesn't just affect 23.05 RCs, but stable 22.03 releases also.

**Comment by nmav** [comment]
I've updated 22.03 to v14. Please try this one whether it fixes the issue.

**Comment by samcday** [comment]
`opkg` isn't showing any updates available for the tang package from 22.03 repositories, yet.

I noticed you fixed some hashes a couple of hours ago - how long is it expected to take for packaging updates to make their way to the repos?

**Comment by samcday** [comment]
tang `14-8` is available in 22.03.5 package repos and appears to fix this issue (for me on `ipq40xx`/`armv7l` platform, at least). Thanks @nmav!

**Comment by nmav** [comment]
Thank you for confirming. I'm closing now this.
